src/shogun/mathematics/graph/ops/shogun/Add.h

```diff
@@ -54,8 +59,66 @@ namespace shogun
 					auto* CPU_arch = CPUArch::instance();
 					if (CPU_arch->has_avx512f())
 						add_kernel_implementation_avx512f<T>(input1, input2, output, size);
+					else if (CPU_arch->has_avx2())
```

Comment:
@vigsterkr so this isn't sustainable, but we can easily add so many ops that Eigen doesn't support, for example here I added addition for int32 using AVX2 instructions. The problem is just that we need to check for specific type here, i.e. there should be something in the condition checking if avx2 is available and that the type is int32. The float versions are available in AVX (which I guess is safe to assume is a subset of AVX2)